bin/install_uninstall.ml

```diff
@@ -301,9 +300,37 @@ module File_ops_real (W : Workspace) : File_operations = struct
         print_unix_error (fun () -> Path.rmdir dir)
       | Error (e, _, _) ->
         User_message.prerr (User_error.make [ Pp.text (Unix.error_message e) ])
-      | _ -> ()
-
-  let mkdir_p p = Path.mkdir_p p
+      | _ ->
+        User_error.raise
+          [ Pp.textf "Please delete non-empty directory %s manually."
+              (Path.to_string_maybe_quoted dir)
+          ]
+
+  let remove_file_if_exists ?(even_if_it_is_empty_dir = false) dst =
+    if Path.exists dst then
+      print_unix_error (fun () ->
+          match Path.unlink dst with
+          | exception Unix.Unix_error (EISDIR, _, _)
+            when even_if_it_is_empty_dir ->
+            remove_dir_if_exists_and_empty dst
+          | ()
+          | (exception _) ->
+            print_line "Deleting %s" (Path.to_string_maybe_quoted dst))
```

Comment:
Something is a bit odd here:

- we print `Deleting X` after the call to unlink. Usually, we print `Doing ...` before actually doing it
- Why do ignore all other errors?

Reply:
I agree the code is a bit convoluted at the moment. I wrote it like this to avoid seeing 
```
Deleting foo/bar
Deleting empty directory foo/bar
```
Here the first operation fails (because it's a directory and not a file), so we go ahead and try to delete the empty directory. Perhaps, having both messages printed in this case is fine? 

> Why do ignore all other errors?

Other errors do end up being caught and reported elsewhere. We catch `EISDIR` right here because we can recover from this error (by deleting the empty directory).

One way to fix both of the above is to run `Path.is_directory` before calling `remove_file_if_exists` and choose the right function at the call site. That does seem cleaner to me, so I'll give it a try.

Reply:
OK, I think it worked very well -- much simpler in the end!

Reply:
Ack, that makes sense. Agreed the race is not worth worrying about: if the user change the FS in parallel with `dune install`, many things can go wrong anyway.